setdeviceconfig.py

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env python3
+
+import os
+import re
+import yaml
+from shutil import move
+
+
+def split_yaml_params(yaml_raw, params):
+    """ 
+        returns yaml without specified params and yaml from params
+        keeping comments that are above parameters
+    """
+
+    comment_chunk = ""
+    clean_yaml = ""
+    removed_yaml = ""
+    param_regex = re.compile("^\\s*(\\S*):")
+    comment_regex = re.compile("^\\s*#")
+    for line in yaml_raw.splitlines(True):
+        if comment_regex.match(line):
+            comment_chunk += line
+        elif line.split() == "":
+            clean_yaml += comment_chunk
+            comment_chunk = ""
+        else:
+            param_match = param_regex.match(line)
+            if param_match and param_match.group(1) in params:
+                removed_yaml += comment_chunk + line
+                comment_chunk = ""
+            else:
+                clean_yaml += comment_chunk + line
+                comment_chunk = ""
+
+    return clean_yaml, removed_yaml
+
+
+def main():
+    device_config = "/etc/cacophony/device.yaml"
+    device_priv_config = "/etc/cacophony/device-priv.yaml"
+
+    config = "/etc/thermal-uploader.yaml"
+    private_config = "/etc/thermal-uploader-priv.yaml"
+    device_params = ["server-url", "group", "device-name"]
```

Comment:
all the above should be constants at the top of the file (and named in UPPER_CASE_WITH_UNDERSCORES)